src/platform/silabs/wifi/SiWx/WifiInterface.cpp

```diff
@@ -72,6 +71,7 @@ extern "C" {
 #endif
 
 #if CHIP_CONFIG_ENABLE_ICD_SERVER
+#include <app/icd/server/ICDConfigurationData.h>
```

Comment:
So it seems we are moving everything ICD related under this `#if CHIP_CONFIG_ENABLE_ICD_SERVER`, why not also move `#include <app/icd/server/ICDServerConfig.h>`?

Reply:
ICDServerConfig.h is the header that inlucdes the `CHIP_CONFIG_ENABLE_ICD_SERVER` define so it always needs to be included before using the define wether its an icd build or not.

Reply:
Yeah, many files has this header file included but not under this macro like MatterConfig.cpp,demo-ui.cpp, so better take it as separate task instead of adding in the same PR.